File: patterns/java/src/refactoring_guru/patterns/factory_method/example/buttons/HtmlButton.java
package refactoring_guru.patterns.factory_method.example.buttons;

/**
 * EN: HTML button implementation.
 * 
 * RU: Реализация HTML кнопок.
 */
public class HtmlButton implements Button {

    public void render() {
        System.out.println("<button>Test Button</button>");
        onClick();
    }

    public void onClick() {
        System.out.println("Click! Button says - 'Hello World!'");
    }
}

File: patterns/cs/src/RefactoringGuru/FactoryMethod/Example/Buttons/HtmlButton.cs
using System;

namespace RefactoringGuru.FactoryMethod.Example.Buttons
{
    /**
     * EN: HTML button implementation.
     * 
     * RU: Реализация HTML кнопок.
     */
    class HtmlButton : IButton
    {
        public void Render()
        {
            Console.WriteLine("<button>Test Button</button>");
            OnClick();
        }

        public void OnClick()
        {
            Console.WriteLine("Click! Button says - 'Hello World!'");
        }
    }
}
